### PR TITLE
Implement sessions multi process

### PR DESCRIPTION
**Pull request by jrothfeder**
PR tracking merge in to master.

### CODE REVIEW

**Review by themiswang — APPROVED** [review]

File: buildSrc/src/main/java/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.kt [hunk 1]
```diff
@@ -85,7 +85,7 @@ class FirebaseLibraryPlugin : BaseFirebaseLibraryPlugin() {
     android.testServer(FirebaseTestServer(project, firebaseLibrary.testLab, android))
     setupStaticAnalysis(project, firebaseLibrary)
     getIsPomValidTask(project, firebaseLibrary)
-    setupVersionCheckTasks(project, firebaseLibrary)
+    // setupVersionCheckTasks(project, firebaseLibrary)
```
Comment (davidmotson) [inline buildSrc/src/main/java/com/google/firebase/gradle/plugins/FirebaseLibraryPlugin.kt:88]: Is this task broken here?

File: firebase-sessions/src/test/kotlin/com/google/firebase/sessions/api/FirebaseSessionsDependenciesTest.kt [hunk 2]
```diff
@@ -19,7 +19,7 @@ package com.google.firebase.sessions.api
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import com.google.common.truth.Truth.assertThat
 import com.google.firebase.sessions.api.SessionSubscriber.Name.CRASHLYTICS
-import com.google.firebase.sessions.api.SessionSubscriber.Name.PERFORMANCE
+import com.google.firebase.sessions.api.SessionSubscriber.Name.MATT_SAYS_HI
```
Comment (davidmotson) [inline firebase-sessions/src/test/kotlin/com/google/firebase/sessions/api/FirebaseSessionsDependenciesTest.kt:22]: Is this intended to be left in?

File: firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerfRegistrar.java [hunk 3]
```diff
@@ -71,15 +64,13 @@ public List<Component<?>> getComponents() {
...
             .add(Dependency.required(FirebaseApp.class))
-            .add(Dependency.required(FirebaseSessions.class))
             .add(Dependency.optionalProvider(StartupTime.class))
             .add(Dependency.required(uiExecutor))
             .eagerInDefaultApp()
             .factory(
                 container ->
                     new FirebasePerfEarly(
                         container.get(FirebaseApp.class),
-                        container.get(FirebaseSessions.class),
```
Comment (davidmotson) [inline firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerfRegistrar.java:82]: I'm not sure I understand why perf is losing its sessions dep here. Is it just no longer needed early, but is used later?

File: firebase-sessions/src/test/kotlin/com/google/firebase/sessions/api/FirebaseSessionsDependenciesTest.kt (hunk 2 above)
Comment (jrothfeder) [reply firebase-sessions/src/test/kotlin/com/google/firebase/sessions/api/FirebaseSessionsDependenciesTest.kt:22]: This was intentional. We removed perf integration for now to cut scope.

File: firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerfRegistrar.java (hunk 3 above)
Comment (jrothfeder) [reply firebase-perf/src/main/java/com/google/firebase/perf/FirebasePerfRegistrar.java:82]: This was intentional. We removed perf integration for now to cut scope.

**Pull request merged**